BeeJet.Bot/Commands/Handlers/GameManagement/AddGameCommandHandler.cs

```diff
@@ -1,80 +1,91 @@
-            await AddToGameListChannelAsync(game, categoryChannel, gameInfoEmbed);
+            await AddToGameListChannelAsync(game, categoryChannel, gameInfoEmbed, context);
 
-            var channel = await Guild.CreateTextChannelAsync(game.Trim().Replace(" ", "-"), (properties) => properties.CategoryId = categoryChannel.Id);
-            await Context.RespondAsync($"Channel created", ephemeral: true);
+            var channel = await context.Guild.CreateTextChannelAsync(game.Trim().Replace(" ", "-"), (properties) => properties.CategoryId = categoryChannel.Id);
+            await context.DiscordContext.RespondAsync($"Channel created", ephemeral: true);
 
             var permissionOverrides = new OverwritePermissions(viewChannel: PermValue.Deny);
-            await channel.AddPermissionOverwriteAsync(Guild.EveryoneRole, permissionOverrides);
-            var message = await channel.SendMessageAsync($"This is the channel for {game}", embed: gameInfoEmbed.Build());
+            await channel.AddPermissionOverwriteAsync(context.Guild.EveryoneRole, permissionOverrides);
+            var message = await channel.SendMessageAsync($"This is the channel for {game}", embed: gameInfoEmbed?.Build());
+
         }
 
-        private async Task<ICategoryChannel> GetOrCreateCategoryChannelAsync(string categoryName)
+        private async Task<ICategoryChannel> GetOrCreateCategoryChannelAsync(string categoryName, SlashCommandContext context)
         {
-            ICategoryChannel parentChannel = Guild.Channels.OfType<SocketCategoryChannel>().FirstOrDefault(b => b.Name.Equals(categoryName, StringComparison.OrdinalIgnoreCase));
+            ICategoryChannel parentChannel = context.Guild.Channels.OfType<SocketCategoryChannel>().FirstOrDefault(b => b.Name.Equals(categoryName, StringComparison.OrdinalIgnoreCase));
             if (parentChannel == null)
             {
-                parentChannel = await Guild.CreateCategoryChannelAsync(categoryName);
+                parentChannel = await context.Guild.CreateCategoryChannelAsync(categoryName);
             }
             return parentChannel;
         }
 
-        private async Task AddToGameListChannelAsync(string game, ICategoryChannel categoryChannel, EmbedBuilder gameInfoEmbed)
+        private async Task AddToGameListChannelAsync(string game, ICategoryChannel categoryChannel, EmbedBuilder gameInfoEmbed, SlashCommandContext context)
         {
-            var gameListChannel = await AddOrGetGameListChannelAsync(categoryChannel);
+            var gameListChannel = await AddOrGetGameListChannelAsync(categoryChannel, context);
             var builder = new ComponentBuilder()
-                .WithButton("Join", GameManagementCommandSource.JointButtonId, ButtonStyle.Success)
-                .WithButton("Leave", GameManagementCommandSource.LeaveButtonId, ButtonStyle.Danger);
+                .WithButton("Join", JointButtonId, ButtonStyle.Success)
```

Comment:
Zouden we Join en Leave anders niet in een constante moeten doen?

Reply:
Join is de tekst van de knop